src/components/sbb-header/sbb-header.tsx

```diff
@@ -20,6 +20,80 @@ export class SbbHeader {
    */
   @Prop({ reflect: true }) public expanded = false;
 
+  /** The element's id or the element on which the scroll listener is attached. */
+  @Prop() public scrollOrigin: string | HTMLElement | Document = document;
+
+  /** Whether the header should hide and show on scroll. */
+  @Prop({ reflect: true }) public hideOnScroll = false;
+
+  @State() private _headerOnTop = true;
+
+  @Element() private _element!: HTMLElement;
+
+  private _scrollElement: HTMLElement | Document;
+
+  private _lastScroll = 0;
+
+  /** If `hideOnScroll` is set, checks the element to hook the listener on, and possibly add it.*/
+  public componentDidLoad(): void {
+    if (typeof this.scrollOrigin === 'string') {
+      this._scrollElement = document.getElementById(this.scrollOrigin);
+      if (!this._scrollElement) {
+        return;
+      }
+    } else {
+      this._scrollElement = this.scrollOrigin;
+    }
+
+    const scrollFn: () => void = this.hideOnScroll
+      ? this._scrollListener
+      : this._scrollShadowListener;
+    this._scrollElement.addEventListener('scroll', scrollFn.bind(this), { passive: true });
+  }
+
+  /** Sets the correct value for `scrollTop`, then:
+   * - apply the shadow if the element/document has been scrolled down;
+   * - hides the header, remove the shadow and possibly close any open menu on the header if it is not visible anymore;
+   * - shows the header and re-apply the shadow if the element/document has been scrolled up.
+   */
+  private _scrollListener(): void {
+    const currentScroll = document.documentElement.scrollTop || document.body.scrollTop; // Get current scroll value
+    this.shadow = currentScroll !== 0;
+    const header = this._element.shadowRoot.firstElementChild as HTMLSbbHeaderElement;
+    // check if original header position has been scrolled out
+    if (currentScroll > this._element.offsetHeight) {
+      this._headerOnTop = false;
+      if (currentScroll > 0 && this._lastScroll <= currentScroll) {
+        this.shadow = false;
+        this._element.style.setProperty(
+          '--sbb-header-position',
+          '-' + this._element.offsetHeight + 'px'
+        );
+        (this._element.querySelector('sbb-menu') as HTMLSbbMenuElement)?.close();
```

Comment:
The menu doesn't have to be inside an sbb-header tag/slot, it can also be outside. Maybe check via triggers in the menu? @dauriamarco and @kyubisation what do you think?

Reply:
BTW. it can have more than one menu which should be closed.

Reply:
The story speaks about to 'close open menus (attached to buttons in the header) on scrolling', so I assume they are in the slot; I also added a sbb-button with an attached menu which opens together with the one that is already there, and using something like this close them both:

`(this._element.querySelectorAll('sbb-menu') as NodeListOf<HTMLSbbMenuElement>)?.forEach((e: HTMLSbbMenuElement) => e.close());`

What do you think?

Reply:
discussed in meeting, after marco's PR is merged, check if there is any aria-controls and aria-expanded on it and close the corresponding menus